Whiskey/Functions/New-WhiskeyContext.ps1

```diff
@@ -6,29 +6,54 @@ function New-WhiskeyContext
+    * Sets the version number to "0.0.0".
+
+    ## Whiskey.Context
+
+    The `Whiskey.Context` object has the following properties. ***Do not use any property not defined below.*** Also, these properties are ***read-only***. If you write to them, Bad Things (tm) could happen.
+
+    * `BuildMetadata`: a `Whiskey.BuildInfo` object representing build metadata provided by the build server.
+    * `BuildRoot`: a `System.IO.DirectoryInfo` object representing the directory the YAML configuration file is in.
     * `ByBuildServer`: a flag indicating if the build is being run by a build server.
     * `ByDeveloper`: a flag indicating if the build is being run by a developer.
-    * `BuildMetadata`: metadata information of the current build on a build server.
-    * `ConfigurationPath`: the absolute path to the YAML file passed via the `ConfigurationPath` parameter.
     * `Environment`: the environment the build is running in.
-    * `OutputDirectory`: the path to a directory where build output, reports, etc. should be saved. This directory is created for you.
-    * `PipelineName`: the currently running task pipeline.
-    * `Publish`: a flag indicating if the PublishTasks pipeline should run.
-    * `RunMode`: the current run mode of the build, `Build`, `Clean`, or `Initialize`. Defaults to `build`.
-    * `TaskName`: The currently running task.
+    * `OutputDirectory`: a `System.IO.DirectoryInfo` object representing the path to a directory where build output, reports, etc. should be saved. This directory is created for you. 
+    * `ShouldClean`: a flag indicating if the current build is running in clean mode. 
+    * `ShouldInitialize`: a flag indicating if the current build is running in initialize mode.
     * `Temp`: the temporary work directory for the current task.
-    * `Version`: a `SemVersion.SemanticVersion` object representing the semantic version to use when building the application. This object has two extended properties: `Version`, a `Version` object that represents the semantic version with all pre-release and build metadata stripped off; and `ReleaseVersion` a `SemVersion.SemanticVersion` object with all build metadata stripped off.
+    * `Version`: a `Whiskey.BuildVersion` object representing version being built (see below).
+
+    Any other property is considered private and may removed, renamed, and/or reimplemented at our discretion without notice.
+
+    ## Whiskey.BuildInfo
+
+    The `Whiskey.BuildInfo` object has the following properties.  ***Do not use any property not defined below.*** Also, these properties are ***read-only***. If you write to them, Bad Things (tm) could happen.
+
+    * `BuildNumber`: the current build number. This comes from the build server. (If the build is being run by a developer, this is always "0".) It increments with every new build (or should). This number is unique only to the current build job.

```

Comment:
I used these words on purpose so that users know this is the number that represents the per-build incremented number used by the underlying build server.